manipvortex/manipulator.go

```diff
@@ -621,7 +626,7 @@ func (m *vortexManipulator) backgroundSync(ctx context.Context) {
 			}
 
 			retryCtx, cancel := context.WithDeadline(ctx, t.Deadline)
-			cancel()
+			defer cancel()
 
 			if err := m.commitUpstream(retryCtx, t.Method, t.mctx, t.Object); err != nil {
 				m.RUnlock()
```

Comment:
This is where we need to deal with errors. If the upstream sends Locked or TimeOut, we need to enter a retry loop, up to the deadline. If the retry loop succeeds, then we kick in the rate limiter.

Reply:
But this is not always the case, what If its `context.Background()` (which is what I use in dispatcher) and upstream never returns and all the objects are queued in the `transactionQueue` up until the `transactionQueueLength`.

Reply:
So when the backend comes up, the queue will empty at a rate they can't handle.

Reply:
Eventually, there can be rate limiting here, but the rate limiting is more complex than a static leaky bucket. In reality, since this is in a loop, the backend will automatically rate limit it since multiple of them will happen at the same time and we will just see increased delays. I don't think that actually limiting the traffic here has any benefit.

Reply:
Ok. In that case, I will remove it.

Reply:
@sibicramesh I realized that we need error handling here. If the error returned is because the backend is locked, we cannot continue here, but we need to have retries.